src/EditModel/event-program/tracker-program/program-stages/ProgramStageList.js

```diff
@@ -153,6 +153,8 @@ class ProgramStageList extends Component {
         }
     }
 
+    // here you need to get d2 and pass it as a `d2` prop to sharing
+    // dialog, to do that you need to get d2 in here
```

Comment:
Here I ran into the error you mentioned where it just "hangs". I think that is because of RxJS swallowing the error, but most likely I think the error has to do with an incorrect prop is sent to the SharingDialog.